craft_store/publisher/_response.py

```diff
@@ -0,0 +1,174 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Response models for the publisher gateway."""
+
+import datetime
+import enum
+from typing import Annotated, Any
+
+import annotated_types
+import pydantic
+
+from craft_store.models import _base_model
+from craft_store.models._charm_model import CharmBaseModel as Base
+from craft_store.models._charm_model import ResourceModel as Resource
+from craft_store.models._common_list_releases_model import PackageModel as Package
+from craft_store.models._common_list_releases_model import (
+    ProgressiveModel as Progressive,
+)
+from craft_store.models.error_model import ErrorModel as Error
+
+Length = Annotated[int, annotated_types.Ge(0)]
+"""An integer that represents the length of an object (non-negative)."""
+
+
+RevisionNumber = Annotated[int, annotated_types.Ge(1)]
+"""A store revision number."""
+
+
+Sha3_384 = Annotated[
+    str, annotated_types.Len(96, 96), pydantic.Field(pattern=r"[0-9a-fA-F]{96}")
+]
+"""A hexadecimal string containing a sha3-384 digest."""
+
+
+class SnapConfinement(enum.Enum):
```

Comment:
Is the snap-related data from snapcraft's source code?

Reply:
No, it's from [the Publisher gateway docs](https://api.staging.charmhub.io/docs/default.html#list_revisions), specifically the JSON schema which states:

```
"confinement": {
    "enum": [
        "strict",
        "classic",
        "devmode"
    ],
    "type": "string"
},
```